src/OptimizelyWrapper.php

```diff
@@ -5,30 +5,56 @@
 
 class OptimizelyWrapper
 {
+    const TYPE_IS_FEATURE_ENABLED = "isFeatureEnabled";
+    const TYPE_ACTIVATE = "activate";
+
     private $datafile;
     private $optimizely;
 
     public function isFeatureEnabled(string $experiment, string $userID, $params = []) : bool
     {
+        return $this->cacheOptimizelyResponse($experiment, $userID, $params, self::TYPE_IS_FEATURE_ENABLED);
+    }
+
+    public function activate(string $experiment, string $userID, $params = []) : string
+    {
+        return $this->cacheOptimizelyResponse($experiment, $userID, $params, self::TYPE_ACTIVATE);
+    }
+
+    public function track($event, $userID, $params = [])
+    {
+        $this->optimizely = app()->make(Optimizely::class);
+        return $this->optimizely->track($event, $userID, $params);
+    }
+
+    /**
+     * @return Optimizely\Optimizely
+     * In case for some reason we want to access the optimizely instance directly.
+     */
+    public function getOptimizelyInstance()
+    {
+        $this->optimizely = app()->make(Optimizely::class);
+        return $this->optimizely();
+    }
+
+    private function cacheOptimizelyResponse($experiment, $userID, $params, $type) {
         $this->datafile = app()->make(Datafile::class)->get();
         $this->optimizely = app()->make(Optimizely::class);
 
-        $cacheKey = 'isFeatureEnabled-' . $experiment . $userID . md5(json_encode($params)) . md5($this->datafile);
+        $cacheKey = $type . '-' . $experiment . $userID . md5(json_encode($params)) . md5($this->datafile);
 
         if (!Cache::has($cacheKey))
         {
-            $variant = $this->optimizely->isFeatureEnabled($experiment, $userID, $params);
+            if ($type === self::TYPE_IS_FEATURE_ENABLED) {
```

Comment:
You could do 
`$variant = $this->optimizely->$type($experiment, $userId, $params);` here

Reply:
👍

Reply:
I think it's good enough. Feel free to merge at will @Speedrockracer, or adjust according to @Robertbaelde's review.